aimmo-game/tests/functional/test_health_pickups_and_effects.py

```diff
@@ -16,27 +19,63 @@ def setUp(self):
         self.game = MockWorld()
         self.game.game_state.add_avatar(1, None, Location(0, 0))
         self.cell = self.game.game_state.world_map.get_cell(Location(1, 0))
+        self.initial_health = self.game.avatar_manager.get_avatar(1).health
 
     def test_health_pickups_and_effects_apply_default(self):
+        """
+        HealthPickups without any parameter provided.
+        """
         self.cell.pickup = HealthPickup(self.cell)
 
         self.game.turn_manager._run_single_turn()
 
         self.assertEqual(self.cell.avatar, self.game.avatar_manager.get_avatar(1))
-        self.assertEqual(self.cell.avatar.health, 5 + HEALTH_RESTORE_DEFAULT)
+        self.assertEqual(self.cell.avatar.health, self.initial_health + HEALTH_RESTORE_DEFAULT)
 
-    def test_health_pickups_and_effects_apply_custom(self):
-        custom_value = 10.5
-        self.assertNotEqual(int(round(custom_value)), HEALTH_RESTORE_DEFAULT)
-        self.cell.pickup = HealthPickup(self.cell, custom_value)
+    @given(st.integers(1, 100))
+    def test_health_pickups_and_effects_apply_custom_integers(self, restore_value):
+        """
+        HealthPickups with explicit integer parameter provided.
+        """
+        self.setUp()
+        self.cell.pickup = HealthPickup(self.cell, restore_value)
 
         self.game.turn_manager._run_single_turn()
+        self.assertEqual(self.cell.avatar, self.game.avatar_manager.get_avatar(1))
+
+        if self.initial_health + restore_value > 100:
+            expected_result_health = 100
+        else:
+            expected_result_health = self.initial_health + restore_value
+
+        self.assertEqual(self.cell.avatar.health, expected_result_health)
 
+    @given(st.floats(1, 100))
```

Comment:
can't you use the HEALTH_RESTORE_MAX value here?

Reply:
:+1: !!!